### PR TITLE
:bug: Update postMessage to call correct origin

### PR DESCRIPTION
**Pull request by jmelberg-okta**
### Description
Fixes an issue where the `iframe` triggering the fingerprint request was receiving an invalid origin. This was caused by attempting to `postMessage` to an incorrect origin.

#### Resolves
[OKTA-163625](https://oktainc.atlassian.net/browse/OKTA-163625)

### CODE REVIEW

**Comment by jmelberg-okta** [comment]
@mauriciocastillosilva-okta - Great question. I've updated this PR to ensure the `iframe` has loaded before attempting the `postMessage`. This will ensure we do not miss the initial message from the `iframe` source.

File: lib/clientBuilder.js [hunk 1]
```diff
@@ -244,16 +245,17 @@ proto.fingerprint = function(options) {
...
+          }), sdk.options.url);
+        });
       }
     }
   }
-  oauthUtil.addListener(window, 'message', listener);
-
-  iframe.src = sdk.options.url + '/auth/services/devicefingerprint';
   document.body.appendChild(iframe);
+  oauthUtil.addListener(window, 'message', listener);
```
Comment (mauriciocastillosilva-okta) [inline lib/clientBuilder.js]: I have the same or similar question: so when we get to this line, the iframe has set its ```src``` attribute and it has been attached to the page. Is there a chance then that we miss the first 'message' if the page loads fast?

In the previous version we had the listener before setting the src and before attaching the iframe to the body of the page.

File: lib/clientBuilder.js [hunk 2]
```diff
@@ -244,16 +245,17 @@ proto.fingerprint = function(options) {
     }
     if (msg.type === 'FingerprintServiceReady') {
       if (iframe.contentWindow) {
-        iframe.contentWindow.postMessage(JSON.stringify({
-          type: 'GetFingerprint'
-        }), sdk.options.url);
+        iframe.addEventListener('load', function() {
```
Comment (mauriciocastillosilva-okta) [inline lib/clientBuilder.js]: is this the actual fix or is it a combination of this `load` event listener and the order of configuring and attaching the iframe?

File: lib/clientBuilder.js [hunk 3]
```diff
@@ -243,11 +243,9 @@ proto.fingerprint = function(options) {
       return deferred.resolve(msg.fingerprint);
     }
     if (msg.type === 'FingerprintServiceReady') {
-      if (iframe.contentWindow) {
-        iframe.contentWindow.postMessage(JSON.stringify({
-          type: 'GetFingerprint'
-        }), sdk.options.url);
-      }
+      e.source.postMessage(JSON.stringify({
```
Comment (mauriciocastillosilva-okta) [inline lib/clientBuilder.js]: is ```e.source``` always defined? Or should we check for undefined just in case?

Comment (jmelberg-okta) [reply lib/clientBuilder.js]: It appears to be available on every [`Event`](https://developer.mozilla.org/en-US/docs/Web/API/MessageEvent/source).

The only case where it _could_ be `undefined` is if the event itself is `underfined`, which will get caught [here](https://github.com/okta/okta-auth-js/blob/0a5b9d87eb43535ab0fba605cbc3385f642f0551/lib/clientBuilder.js#L231).

Comment (mauriciocastillosilva-okta) [reply lib/clientBuilder.js]: Sounds good. I also found this info about [the dispatched event](https://developer.mozilla.org/en-US/docs/Web/API/Window/postMessage#The_dispatched_event), which as you said, means we will have a ```source```

**Pull request merged**